src/library/refresher.py

```diff
@@ -253,14 +255,10 @@ def clean_datasets(stale_datasets, changed_datasets):
                     clean_containers_by_id(blob_service_client, file_id, containers=[
                                            config['CLEAN_CONTAINER_NAME']])
 
-                # remove from all solr collections
-                for core_name in solr_cores:
-                    try:
-                        solr_cores[core_name].delete(
-                            q='iati_activities_document_id:' + file_id)
-                    except:
-                        logger.warn('Failed to remove changed docs from solr with hash: ' +
-                                    file_hash + ' and id: ' + file_id + ' from core with name ' + core_name)
+                # Don't remove from any solr collections!
```

Comment:
@odscjames after reviewing validator-web, it occurred to me that right here I think we want to unset `solrize_end`, but retain `last_solrize_end`. Otherwise validator-web is just printing "Old version" for every document, because every document that's been solrized will have both solrize_end and last_solrize_end set. But if we set solrize_end to null here, then validator-web can check for solrize_end === null & last_solrize_end, and correctly identify old versions.

Reply:
> "it occurred to me that right here I think we want to unset solrize_end, but retain last_solrize_end."

Does this not happen anyway? clean_datasets function with any value for changed_datasets var is only called in sync_documents function and there as part of building changed_datasets var the function db.insertOrUpdateDocument is called - and that does wipe all the stage start/end fields if the hash is different (but not last_solrize_end).

Reply:
Yes I think you're correct. Should be all good, then.